### PR TITLE
Adds custom db_copy_to method for MS SQL

### PR DESCRIPTION
**Pull request by edgararuiz**
Fixes https://github.com/tidyverse/dplyr/issues/3174

It adds a custom `db_copy_to()` method for MS SQL connections that is a copy of the base `odbc` `db_copy_to()` but it prefixes the `##` if needed and executes the `dbWriteTable()` command with a `temporary` argument set to `FALSE`:

Test results: 

```
> test <- copy_to(con, mtcars, "mtcars2")
Created a temporary table named: ##mtcars2
> test
# Source:   table<##mtcars2> [?? x 12]
# Database: Microsoft SQL Server 14.00.1000[dbo@DESKTOP-10DBTVP\SQLEXPRESS/master]
           row_names   mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb
               <chr> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
 1         Mazda RX4  21.0     6 160.0   110  3.90 2.620 16.46     0     1     4     4
 2     Mazda RX4 Wag  21.0     6 160.0   110  3.90 2.875 17.02     0     1     4     4
 3        Datsun 710  22.8     4 108.0    93  3.85 2.320 18.61     1     1     4     1
 4    Hornet 4 Drive  21.4     6 258.0   110  3.08 3.215 19.44     1     0     3     1
 5 Hornet Sportabout  18.7     8 360.0   175  3.15 3.440 17.02     0     0     3     2
 6           Valiant  18.1     6 225.0   105  2.76 3.460 20.22     1     0     3     1
 7        Duster 360  14.3     8 360.0   245  3.21 3.570 15.84     0     0     3     4
 8         Merc 240D  24.4     4 146.7    62  3.69 3.190 20.00     1     0     4     2
 9          Merc 230  22.8     4 140.8    95  3.92 3.150 22.90     1     0     4     2
10          Merc 280  19.2     6 167.6   123  3.92 3.440 18.30     1     0     4     4
# ... with more rows
```

### CODE REVIEW

**Comment by edgararuiz** [comment]
PTAL @hadley

**Comment by hadley** [comment]
Why is this needed? `db_copy_to()` already calls `db_write_table()`.

**Comment by edgararuiz** [comment]
I think currently `db_copy_to.OdbcConnection` is called first, and it uses `dbWriteTable()` instead of `db_write_table()`. That's why I figured to create the custom method `db_copy_to()` for MS SQL.  

If you prefer, I could have `db_copy_to.OdbcConnection` call `db_write_table()` instead, and push the custom code to a new `db_write_table.MS SQL Server`  method, and get rid of the MS SQL `db_copy_to()`

**Comment by hadley** [comment]
Yes, that would be better, I think.

**Comment by hadley** [comment]
I think `db_copy_to.DBIConnection()` also needs a change?  `db_write_table(con, table, types = types, values = values, temporary = temporary)` should now overwrite `table`

**Comment by hadley** [comment]
Doh, you did that already.

File: R/db-odbc-mssql.R [hunk 1]
```diff
@@ -126,6 +123,28 @@
   name
 }
 
+#' @export
+`db_write_table.Microsoft SQL Server`  <- function(con, table, types, values, temporary = TRUE, ...) {
+
+  # check that name has prefixed '##' if temporary
```
Comment (hadley) [inline R/db-odbc-mssql.R]: Can you please extract this repeated code into a function?

**Comment by edgararuiz** [comment]
PTAL @hadley 

Updating `db_copy_to.DBIConnection()` required making any `db_write_table()` method return the `table` name.

**Pull request merged**